### PR TITLE
Bug when trying to use both 'href' and 'tooltip'

### PR DESCRIPTION
**Pull request by faizan-khan-iit**
This PR covers Animint issues #100 and #155

### CODE REVIEW

File: inst/htmljs/animint.js [hunk 1]
```diff
@@ -1696,8 +1696,7 @@ var animint = function (to_select, json_file) {
 	  return d["clickSelects.variable"] + " " + d["clickSelects.value"];
 	};
       }
-      elements.text("")
-        .append("svg:title")
```
Comment (faizan-khan-iit) [inline inst/htmljs/animint.js]: What is `elements.text("")` doing here? Removing it seems to solve the problem...

**Comment by faizan-khan-iit** [comment]
After this minor change, everything seems to work fine.

Here is the example from #155 - [Worldbank example](http://bl.ocks.org/faizan-khan-iit/449b84b2e6b55ca5e39ea9aaad97e27e)
And here is the example from #100 - [Visualization of Ridge Hike](http://bl.ocks.org/faizan-khan-iit/0b6f5c86443774f7c6e0d83708f9ff06)

But the wercker build is stuck for some reason. Maybe ignore it since the Travis build is passing?

@tdhock @cpsievert @kferris10 Could one of you please confirm?

**Comment by tdhock** [comment]
Thanks for working on this Faizan. Before starting to code various solutions, can you please add at least one test case that fails?

**Comment by tdhock** [comment]
Also please add a test case for the bug in your approach that you found.

**Comment by faizan-khan-iit** [comment]
@tdhock Sorry about that. I found out about the bug just before I had to leave so I thought to put in a comment so that you people do not waste your time.

So in the initial bug that we were trying to fix, `<circle>`  was getting overwritten by `elements.text("")` when using tooltip with href since the both `circle` and `title` were children of anchor tag `a`. So I thought removing `elements.text("")` should do the job. Which led to [this viz](http://bl.ocks.org/faizan-khan-iit/449b84b2e6b55ca5e39ea9aaad97e27e), as I mentioned above.

Initially it works fine, but while interacting with the points, the code keeps appending `title` elements one after the other, which is the bug I noticed. If you navigate to the above viz, initially all is good. Notice the Canada and US points in blue in the top left of the plot. But once you hide all the points excluding North America, you will see that their tooltip labels are now Armenia and Albania (that's what I get). The titles get updated every time I hide or unhide elements, and the topmost value is displayed every time.

Hence I currently remove `<tilte>` every time an update takes place. This results in [this viz](http://bl.ocks.org/faizan-khan-iit/6a483d9bd03f2c939ceb11ecc294d22c), which seems fine for all the testing I have put it through.

Hopefully this explains the changes and tests that I made.

Still the tooltip test gets stuck somewhere. I will see what is causing this

**Comment by tdhock** [comment]
phantomjs has problems when there are several elements with the same ID -- is that the problem?

**Comment by faizan-khan-iit** [comment]
@tdhock I don't think so. The HTML is quite straight-forward and there are no ID clashes. I specify the IDs to be clicked [here](https://github.com/tdhock/animint/pull/156/files#diff-917aab0dfef94d843e17604a14008ed2R139).

I tried to find out where the execution gets stuck, and I found out it gets stuck here, at [the first getHTML() call](https://github.com/tdhock/animint/pull/156/files#diff-917aab0dfef94d843e17604a14008ed2R149).

Should I write a different test if this doesn't work?

**Comment by tdhock** [comment]
try removing non-alphanumeric characters from the ids (or convert them to underscores). I have a feeling that `(&)` characters are the problem.

**Comment by faizan-khan-iit** [comment]
@tdhock Great observation! Once I filtered out the non-alphanumeric chars, the tests pass locally using `tests_run(filter="tooltip")`. 

But the bad news is that they get stuck when running `tests_run(filter="renderer1")`, like in the builds. 

I separated the tests which are causing the problem. Probably ignore these for now? We could check this separately using firefox locally...

**Comment by tdhock** [comment]
Try renaming that test file to `test-renderer4-tooltip.R` and then adding `TEST_SUITE=renderer4` to the travis and wercker builds.

**Comment by faizan-khan-iit** [comment]
travis seems fine with the changes but looks like wercker is still stuck with the new tests.

**Comment by faizan-khan-iit** [comment]
@tdhock Could you please restart again. It failed before starting the tests...

**Comment by tdhock** [comment]
looks like there are still problems. can you double-check everything on that page to make sure there are no duplicate id values, and no id values with non-alphanumeric characters? you may even want to try deleting `_-` characters to see if that helps, and make sure that the id does not start with a number. Not sure why all of this is necessary, I guess it is some bug in phantomjs...

**Comment by faizan-khan-iit** [comment]
@tdhock Looks like wecker build failed again without starting the tests..

I cut the region names to the first three letters. So these are the html id's we are working with now https://travis-ci.org/tdhock/animint/jobs/142332160#L2663

I will remove these once we complete the wercker build. Should I try and remove the rest of the underscores too? I would probably need to change the JS code, which may cause unexpected problems in the other tests.

Also there are no conflicts whatsoever in the rendered HTML. The only point worth noting is that the `svg` and `label` elements under these IDs mentioned have IDs with "_svg" and "_label" appended to them. Eg.
Legend ID: plot_ex_region_variable_eas
SVG ID: plot_ex_region_variable_eas_svg
Label ID: plot_ex_region_variable_eas_label

But that should not be problem since there are no clashes.

**Comment by tdhock** [comment]
@cpsievert can you please give @faizan-khan-iit permissions to restart your wercker build?

faizan I don't understand why the wercker tests fail... can you try simplifying the test or using a different data set?

also while we debug this, feel free to start working on something else in another branch...

**Comment by faizan-khan-iit** [comment]
@cpsievert Thanks!

@tdhock Looks like the new tests work well with wercker. As for the older tests, I have not deleted those yet, but removed them from the test suites. If you approve of this, I could update the NEWS and DESCRIPTION.

A minor note: I had to install ggplot2 separately for wercker. Don't know why but it was not installing ggplot2 from our PR...

**Comment by tdhock** [comment]
+1 looks good to me, go ahead and increase the version in NEWS and DESCRIPTION, then you can merge with master

**Pull request merged**